sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1969,7 +1969,8 @@ trait SupportsSubquery extends LogicalPlan
 case class CollectMetrics(
     name: String,
     metrics: Seq[NamedExpression],
-    child: LogicalPlan)
+    child: LogicalPlan,
+    dataframeId: Long)
```

Comment:
@amaliujia  when I execute the following command:

```
build/mvn clean install -pl connector/connect/server -am -DskipTests
mvn test -pl connector/connect/server 
```
The following test failures will occur:

```
- Test observe *** FAILED ***
  == FAIL: Plans do not match ===
  !CollectMetrics my_metric, [min(id#0) AS min_val#0, max(id#0) AS max_val#0, sum(id#0) AS sum(id)#0L], 0   CollectMetrics my_metric, [min(id#0) AS min_val#0, max(id#0) AS max_val#0, sum(id#0) AS sum(id)#0L], 53
   +- LocalRelation <empty>, [id#0, name#0]                                                                 +- LocalRelation <empty>, [id#0, name#0] (PlanTest.scala:179) 
```
It seems that the failure is due to the differences in `dataframeId` when comparing plans.

For this, I've created SPARK-45357

Reply:
https://github.com/apache/spark/blob/ab92cae78e3cdf58ba96b0b98e7958287c2d5cd1/connector/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala#L958-L1017

For this test case, should we ignore the comparison of dataframeId?

Reply:
GA can pass the test now, which seems to be because the test order makes sparkTestRelation.id exactly 0. However, the test arrangement order of Maven is different from sbt, so sparkTestRelation.id is not 0.

Reply:
omitting df id in the comparison of this test case makes sense to me.